### PR TITLE
Remove dependency from Guided Experience to Entity Text

### PR DESCRIPTION
**Pull request by haoranpb**
#### Summary 

Remove dependency from Guided Experience to Entity Text

The Marketing Text Tour will show up on Eval companies, so no need to filter anymore. The Marketing Text FactBox will always show up in Eval Company (tho additional setup might be needed to use the functionality)

#### Work Item(s) 
Fixes [AB#495295](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/495295)

### CODE REVIEW

File: src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al [hunk 1]
```diff
@@ -41,4 +41,15 @@ codeunit 3728 "Checklist Banner"
     internal procedure OnOpenChecklistBannerPage(var SkipWelcomeState: Boolean; IsEvaluationCompany: Boolean)
     begin
     end;
+
+    /// <summary>
+    /// Integration Event to provide the ability to filter on the checklist items before it is loaded to the checklist banner.
+    /// Do NOT use complex logic in the subscriber as it will impact the performance of opening company.
```
Comment (grobyns) [inline src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al]: seems like we shouldn't have this event at all... Why create the possibility of things going wrong?

File: src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al [hunk 2]
```diff
@@ -41,4 +41,15 @@ codeunit 3728 "Checklist Banner"
...
+    /// </summary>
+    /// <param name="SkipChecklistItem">Whether to skip the checklist item or not </param>
+    /// <param name="SpotlightTourType">Specifies the checklist's Spotlight Tour type if any</param>
+    [IntegrationEvent(false, false)]
+    internal procedure OnBeforeAddChecklistItemToBuffer(var SkipChecklistItem: Boolean; SpotlightTourType: Enum "Spotlight Tour Type")
```
Comment (darjoo) [inline src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al]: Missing the Checklist Item itself. Any subscriber would not know what they are skipping except for the type.

File: src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al (hunk 1 above)
Comment (haoranpb) [reply src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al]: Agree, trying a few things to try to get rid of it

File: src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al (hunk 2 above)
Comment (haoranpb) [reply src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al]: Privacy deployment failed, so asked Sam directly.

The factbox is shown when there is read access to entity text, which is granted via `"System Application - View"`

So I guess it's safe to say in Eval Company, the Text Box we are looking for is always there. I vote we just remove the logic as the PR's current state

**Review by darjoo — APPROVED** [review]

File: src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al (hunk 1 above)
Comment (haoranpb) [reply src/System Application/App/Guided Experience/src/Checklist Banner/ChecklistBanner.Codeunit.al]: Removed event after swtiching approach

**Review by grobyns — APPROVED** [review]

**Pull request merged**